core/prop/ranks.go

```diff
@@ -0,0 +1,89 @@
+package prop
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/otrego/clamshell/core/movetree"
+)
+
+// ranksConv converts the rank properties BR and WR.
+var ranksConv = &SGFConverter{
+	Props: []Prop{"BR", "WR"},
+	Scope: RootScope,
+	From: func(n *movetree.Node, prop string, data []string) error {
+		if len(data) != 1 {
+			return fmt.Errorf("%s property requires exactly 1 value, but had %d", prop, len(data))
+		}
+
+		rank := data[0]
+		err := isValidRank(rank)
+		if err != nil {
+			return err
+		}
+
+		if n.GameInfo == nil {
+			// For safety, make sure to set create gameinfo if it doesn't exist.
+			n.GameInfo = &movetree.GameInfo{}
+		}
+		if prop == "BR" {
+			n.GameInfo.BlackRank = rank
+		} else {
+			n.GameInfo.WhiteRank = rank
+		}
+		return nil
+	},
+	To: func(n *movetree.Node) (string, error) {
+		if n.GameInfo == nil {
+			return "", nil
+		}
+
+		var out strings.Builder
+		if n.GameInfo.BlackRank != "" {
+			rank := n.GameInfo.BlackRank
+			err := isValidRank(rank)
+			if err != nil {
+				return "", err
+			}
+			out.WriteString("BR[" + rank + "]")
+		}
+
+		if n.GameInfo.WhiteRank != "" {
+			rank := n.GameInfo.WhiteRank
+			err := isValidRank(rank)
+			if err != nil {
+				return "", err
+			}
+			out.WriteString("WR[" + rank + "]")
+		}
+
+		if out.String() == "" {
+			return "", nil
+		}
+		return out.String(), nil
+	},
+}
+
+func isValidRank(rank string) error {
```

Comment:
I just re-read over the spec. https://www.red-bean.com/sgf/properties.html#BR

BR and WR are just simpletext. There are conventions, but I don't think we can guarantee the format. So, rather than being restrictive, I think we should probably remove the intensive validation and just do the basic validation.